src/snakemake/deployment/__init__.py

```
@@ -0,0 +1,103 @@
+        for kind, plugin in self.plugins.items():
+            if kind in global_variables:
+                raise WorkflowError(
+                    f"The software deployment plugin {plugin.name} provides an "
+                    f"invalid kind {kind}, which is already reserved as a global "
+                    "variable in snakemake."
+                )
+
+        for kind, plugin in self.plugins.items():
+            def make_factory(plugin_val, kind_val):
+                def factory(*args, within: EnvSpecBase, **kwargs):
+                    if args:
```

Ruff (0.14.6)

Function definition does not bind loop variable `args`

(B023)